packages/create-amplify/src/package_manager.ts

```diff
@@ -0,0 +1,58 @@
+export type PackageManagerName =
+  | 'npm'
+  | 'yarn-classic'
+  | 'yarn-modern'
+  | 'pnpm';
+type PackageManagerExecutable = 'npm' | 'yarn' | 'pnpm';
+type PackageManagerBinaryRunner = 'npx' | 'yarn' | 'pnpm';
+type PackageManagerInstallCommand = 'install' | 'add';
+type PackageManagerLockFile =
+  | 'package-lock.json'
+  | 'yarn.lock'
+  | 'pnpm-lock.yaml';
+type PackageManagerInitDefault = ['init', '--yes'] | ['init'];
+export type PackageManager = {
+  name: PackageManagerName;
+  executable: PackageManagerExecutable;
+  binaryRunner: PackageManagerBinaryRunner;
+  installCommand: PackageManagerInstallCommand;
+  lockFile: PackageManagerLockFile;
+  initDefault: PackageManagerInitDefault;
+};
+export type PackageManagers = {
```

Comment:
These should be implementations of PackageManagerController. All of these strings should be internal to the class rather than exposed to other classes that control the behavior

Reply:
Now since we have `NpmPackageManagerController` and other Package Manager Controllers, we need to have some logic to decide which PackageManagerController to use, so I think it's good to keep them under the `package_manager` config